src/linux.rs

```diff
@@ -160,8 +169,7 @@ fn get_desktop_environment() -> DesktopEnvironment {
 	// Useful particularly in the DESKTOP_SESSION=default case.
 	if env_has_var("GNOME_DESKTOP_SESSION_ID") {
 		return DesktopEnvironment::Gnome;
-	}
-	if env_has_var("KDE_FULL_SESSION") {
+	} else if env_has_var("KDE_FULL_SESSION") {
```

Comment:
Note that these are equivalent even if the compiler makes no optimizations. It's because in both version the following is true: If we enter the first "if block", then we will return without checking the second condition. If we don't enter the first "if block" the second condition is going to get checked.

Reply:
In general it's a good idea to avoid making changes in PRs which don't add value, but you can leave like this as it really makes no difference at this point.

Reply:
I was using this library and noticed that it ran really slow, so I looked at the source code and saw all those checks without actually realizing what you're saying, so I just attributed to it without a lot of thought lol. Sorry for that, not my brightest moment. In the end it was kioclient5 that ran slow.